### PR TITLE
feat(passkey): login

### LINKED ISSUE #16: Login: Login with Passkey
As an enduser I want to be able to authenticate my user with passkey.

**Acceptance Criteria**
- [x] Ui checks what kind of authentication methods should be shown depending on the settings of the organization or instance and the configuration the user has
- [x] If passkey is configured challenge is directly started without the user clicking a button
- [x] User is able to abort and choose another authentication method if settings and user configuration does allow

### PR DESCRIPTION
**Pull request by peintnermax**
Closes #16 

This PR implements passkey login for the new UI.

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] Jest unit tests ensure that components produce expected outputs on different inputs.
- [x] Cypress integration tests ensure that login app pages work as expected. The ZITADEL API is mocked.
- [x] No debug or dead code
- [x] My code has no repetitions

### CODE REVIEW

File: apps/login/cypress/integration/login.cy.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    );
+    addStub(
+      "zitadel.user.v2alpha.UserService",
+      "ListAuthenticationMethodTypes",
+      {
+        data: {
+          authMethodTypes: [1], // 1 for password authentication
+        },
+      }
+    );
```
Comment (eliobischof) [inline apps/login/cypress/integration/login.cy.ts:56]: Let's move stubs to beforeEach hooks. Also in verify.cy.ts.

We can nest contexts with describe blocks.
maybe a describe block with password and one with passwordless?
Their tests have common stubs, which we could even reuse in beforeEach.

File: apps/login/mock/initial-stubs/zitadel.settings.v2alpha.SettingsService.json [hunk 2]
```diff
@@ -7,11 +7,40 @@
...
+  },
+  {
+    "service": "zitadel.settings.v2alpha.SettingsService",
+    "method": "GetLoginSettings",
+    "out": {
+      "data": {
+        "settings": {
+          "passkeysType": 1
+        }
+      }
```
Comment (eliobischof) [inline apps/login/mock/initial-stubs/zitadel.settings.v2alpha.SettingsService.json:43]: Should we really initialize the stubs with concrete values?
I'm afraid of implicit hard dependencies if tests rely on the exact values.

File: apps/login/ui/UsernameForm.tsx [hunk 3]
```diff
@@ -34,22 +49,77 @@ export default function UsernameForm() {
...
-      return router.push(
-        `/password?` +
-          new URLSearchParams({ loginName: `${factors.user.loginName}` })
-      );
+  async function setLoginNameAndGetAuthMethods(values: Inputs) {
+    return submitLoginName(values).then((response) => {
+      if (response.authMethodTypes.length == 1) {
+        const method = response.authMethodTypes[0];
+        switch (method) {
+          case 1: //AuthenticationMethodType.AUTHENTICATION_METHOD_TYPE_PASSWORD:
```
Comment (eliobischof) [inline apps/login/ui/UsernameForm.tsx:62]: Does the incommented value not work?

File: apps/login/mock/initial-stubs/zitadel.settings.v2alpha.SettingsService.json (hunk 2 above)
Comment (peintnermax) [reply apps/login/mock/initial-stubs/zitadel.settings.v2alpha.SettingsService.json:43]: I agree for the loginsettings. The rest could stay in there IMHO as the are single use

Comment (peintnermax) [reply apps/login/mock/initial-stubs/zitadel.settings.v2alpha.SettingsService.json]: done

File: apps/login/ui/UsernameForm.tsx (hunk 3 above)
Comment (peintnermax) [reply apps/login/ui/UsernameForm.tsx:62]: The commented code imports other serverside components, that result in an error as this `UsernameForm.tsx` is executed in the browser

File: apps/login/cypress/integration/login.cy.ts (hunk 1 above)
Comment (peintnermax) [reply apps/login/cypress/integration/login.cy.ts]: I've restructured the `login.cy.ts` file 👍

Comment (eliobischof) [reply apps/login/cypress/integration/login.cy.ts]: I reduced the code a bit more https://github.com/zitadel/typescript/pull/33/commits/1bd5ea4e2d7e702d3e0c7b0702d1b377700c0c71

**Review by eliobischof — APPROVED** [review]

**Pull request merged**